packages/backend-common/src/reading/types.ts

```diff
@@ -105,6 +105,23 @@ export type ReadTreeResponseFile = {
   content(): Promise<Buffer>;
 };
 
+export type FromArchiveOptions = {
+  // A binary stream of a tar archive.
+  stream: Readable;
+  // If unset, the files at the root of the tree will be read.
+  // subpath must not contain the name of the top level directory.
+  subpath?: string;
+  // etag of the blob
+  etag: string;
+  // Filter passed on from the ReadTreeOptions
+  filter?: (path: string) => boolean;
+};
+
+export interface IReadTreeResponseFactory {
```

Comment:
We don't use `I` prefixes, so let's just call this `ReadTreeResponseFactory`, and the concrete implementation `DefaultReadTreeResponseFactory`. It shouldn't be a breaking change since the current `ReadTreeResponseFactory` instance has the same signature as the interface and the class itself is not exported.

Reply:
@Rugvip fixed!